Content.Client/Clothing/NightVisionSystem.cs

```diff
@@ -0,0 +1,75 @@
+using Content.Shared.Clothing;
+using Content.Shared.GameTicking;
+using Robust.Client.Player;
+using Robust.Client.Graphics;
+using Content.Client.Inventory;
+using Content.Shared.Inventory.Events;
+
+namespace Content.Client.Clothing;
+
+/// <summary>
+/// Made by BL02DL from _LostParadise
+/// </summary>
+
+public sealed class NightVisionSystem : SharedNightVisionSystem
+{
+    [Dependency] private readonly IOverlayManager _overlayMan = default!;
+    [Dependency] private readonly ILightManager _lightManager = default!;
+    [Dependency] private readonly IPlayerManager _playerManager = default!;
+    [Dependency] private readonly IEntityManager _entityManager = default!;
+
+    private NightVisionOverlay _overlay = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<RoundRestartCleanupEvent>(OnRestart);
+        SubscribeLocalEvent<NightVisionComponent, GotUnequippedEvent>(OnGotUnequipped);
+
+        _overlay = new(this);
+    }
+
+    public NightVisionComponent? GetNightComp()
+    {
+        var playerUid = EntityUid.Parse(_playerManager.LocalPlayer?.ControlledEntity.ToString());
+        var slot = _entityManager.GetComponent<InventorySlotsComponent>(playerUid);
+        _entityManager.TryGetComponent<NightVisionComponent>(slot.SlotData["eyes"].HeldEntity, out var nightvision);
```

Comment:
```suggestion
        _entityManager.TryGetComponent<NightVisionComponent>(slot.SlotData["eyes"].HeldEntity, out var nightvision);
```
Oh this is hardcoded in the "eyes" slot here too. This is incredibly cursed. You shouldn't need a "magic string" here to check only the eyes slot. What if I wanted to make a mech that gives you night vision, or a hardsuit helmet with built-in nightvision?

Reply:
@VMSolidus  What i need to do to fix this?

Reply:
@VMSolidus Please, tell me what i need to do..

Reply:
Refer to what my maintainers are saying